### PR TITLE
puffin: init at 2.1.1

### PR DESCRIPTION
**Pull request by renesat**
## Description of changes

Add tui dashboard for hledger: https://github.com/siddhantac/puffin

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pu/puffin/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+}:
+buildGoModule rec {
+  pname = "puffin";
+  version = "1.7.1";
```
Comment (FliegendeWurst) [inline pkgs/by-name/pu/puffin/package.nix:8]:
```suggestion
  version = "2.1.1";
```

File: pkgs/by-name/pu/puffin/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "siddhantac";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-F9ZGy7oihCrpGueWANjBYUBgWqaesGdmSb0bhsSSYUQ=";
+  };
+
+  vendorHash = "sha256-XfsGXh9nz+QJEAqtw/BfzHhSVje6q0KLsVg1ZQ6fN4U=";
+
+  meta = {
+    description = "A beautiful terminal dashboard for hledger";
```
Comment (FliegendeWurst) [inline pkgs/by-name/pu/puffin/package.nix:20]:
```suggestion
    description = "Beautiful terminal dashboard for hledger";
```

File: pkgs/by-name/pu/puffin/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://github.com/siddhantac/puffin";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ renesat ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/pu/puffin/package.nix:23]:
```suggestion
    mainProgram = "puffin";
    maintainers = with lib.maintainers; [ renesat ];
```

**Review by FliegendeWurst — COMMENTED** [review]
LGTM overall. Maybe update to the latest version.

**Comment by renesat** [comment]
@FliegendeWurst Update and fix suggestions

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/pu/puffin/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  fetchFromGitHub,
+  buildGoModule,
+}:
+buildGoModule rec {
+  pname = "puffin";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "siddhantac";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/pu/puffin/package.nix:12]: Mere suggestion

```suggestion
    repo = "puffin";
```

**Pull request merged**